### PR TITLE
Replace use of properties with features

### PR DESCRIPTION
**Pull request by andy-sweet**
# References and relevant issues
Part of https://github.com/napari/napari/issues/3911

# Description
This replaces use of `properties`, `property_choices`, and `current_properties` with `features` and `feature_defaults`. The motivation for doing this now is to make the latest documentation consistent with existing code and examples on `main` (here and on napari/napari) for the upcoming v0.5 release. As we also plan to deprecate properties in v0.5 (https://github.com/napari/napari/issues/3911), this documentation update is necessary for that.

### CODE REVIEW

**Comment by jni** [comment]
Cool, thanks @andy-sweet! Let me know when you think this is ready for review! 🙏

File: docs/howtos/layers/tracks.md
Comment (andy-sweet) [inline docs/howtos/layers/tracks.md:124]: These docs are incomplete and stale for multiple reasons (including properties/features), so I removed this in favor of link to the two functions in the text above.

File: docs/tutorials/annotation/annotate_points.md
Comment (andy-sweet) [inline docs/tutorials/annotation/annotate_points.md:7]: The changes in this file mostly come from the corresponding PR over on the pointannotator repo: https://github.com/kevinyamauchi/PointAnnotator/pull/6

I tested the example on that repo with napari's `main`.

There are also some other name swaps.

File: docs/tutorials/segmentation/annotate_segmentation.md
Comment (andy-sweet) [inline docs/tutorials/segmentation/annotate_segmentation.md:103]: The code changes come from napari's `examples/annotate_segmentation_with_text.py` example and should be consistent now.

Other changes in this file are mostly word swaps.

File: docs/tutorials/tracking/cell_tracking.md
Comment (andy-sweet) [inline docs/tutorials/tracking/cell_tracking.md:175]: I have not tried upstreaming these changes to https://github.com/quantumjot/btrack/blob/caa56fa82330e4b16b5d28150f9b60ed963165c7/examples/napari_btrack.ipynb

Given their simplicity, I'm fairly confident things should work, but it's probably worth verifying.

**Comment by andy-sweet** [comment]
@jni: I have one last compatibility check to do with the btrack cell-tracking example, but other this is ready for review now.

**Comment by andy-sweet** [comment]
> @jni: I have one last compatibility check to do with the btrack cell-tracking example, but other this is ready for review now.

Check is done.

File: docs/howtos/layers/points.md
Comment (brisvag) [inline docs/howtos/layers/points.md]: maybe a bit confusing for folks unfamiliar with pandas to talk about columns/rows in a dictionary.

**Review by brisvag — COMMENTED** [review]
Other than the specific comment below, I think there's a general overlap between "table" and "dictionary" which might be confusing to new users.

File: docs/howtos/layers/points.md
Comment (jni) [reply docs/howtos/layers/points.md]: Ah, indeed, maybe worth saying that this is converted internally to a data frame with two columns and three rows? (I wouldn't specify pandas necessarily, but maybe.)

File: docs/tutorials/segmentation/annotate_segmentation.md
Comment (jni) [inline docs/tutorials/segmentation/annotate_segmentation.md:358]: I always forget about this. It's so cool. 😃

**Review by jni — APPROVED** [review]
Thanks @andy-sweet, this is excellent! I think @brisvag's minor comment is worth addressing, but I'm also ok with it happening later, so I'm gonna approve this and ready-to-merge it, and if you get to it great and if not we can make an issue. 😂

File: docs/howtos/layers/points.md
Comment (andy-sweet) [reply docs/howtos/layers/points.md]: Agreed this paragraph is mixing things a bit too much. Let me try to fix that.

But I also want to make a distinction between the meaning of `features` (i.e. it's a table) and the type used to provide or implement it (i.e. a `dict` or `DataFrame`). Ideally, we could do that in one place for all layers (see https://github.com/napari/docs/issues/23).

For now, there's a little effort above to describe what the table is in the "### Using the points features table" section. I'll clean that up a little and will mention that the table can be provided as a dictionary where the values are array-likes of the same length.

Comment (andy-sweet) [reply docs/howtos/layers/points.md]: I also (tried to) put a link back to the section that describes the dictionary and table/dataframe relationship in both of these similar paragraphs.

**Comment by andy-sweet** [comment]
I think I addressed the feedback and also fixed a couple of links/cross-references. Please take another quick look before merging!

**Review by brisvag — APPROVED** [review]
Love the changes :)

**Comment by andy-sweet** [comment]
Thanks for the second look. I'll merge this after the weekend unless anyone objects.

File: docs/howtos/layers/tracks.md
Comment (jni) [reply docs/howtos/layers/tracks.md:124]: lmao I came here while trying to resolve merge conflicts with #420 going, "why on earth did this get removed. And no discussion either, wth!"

... Then I found my thumbs-up above 🤣

**Pull request merged**
